### PR TITLE
#1897 Add Kotlin dependencies, plugins, and extensions

### PR DESCRIPTION
**Pull request by kpavlov**
## Issue

Discussion #1897

## Change
Porting [kpavlov/langchain4j-kotlin](https://github.com/kpavlov/langchain4j-kotlin)

Integrated Kotlin support by adding necessary dependencies and plugins to the project. Introduced Kotlin extensions for ChatLanguageModel with corresponding tests. Updated .editorconfig for improved import and properties formatting.

- Kotlin dependencies added as optional, so it will not pollute java projects
- Maven Javadoc plugin replaced with [Dokka](https://kotlinlang.org/docs/dokka-maven.html) in core module in order to include both Java and Kotlin docs
- Upgrade Kotlin to [2.1.0](https://blog.jetbrains.com/kotlin/2024/11/kotlin-2-1-0-released/)

## General checklist

- [x] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [x] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new maven module

- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`

## Checklist for adding new embedding store integration

- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration

- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

### CODE REVIEW

File: langchain4j-core/pom.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
         <Xlint>-options</Xlint>
         <!-- TODO: remove enforcer.skipRules -->
         <enforcer.skipRules>requireUpperBoundDeps</enforcer.skipRules>
+        <maven.javadoc.skip>true</maven.javadoc.skip><!-- use dokka for javadocs -->
```
Comment (langchain4j) [inline langchain4j-core/pom.xml]: Please make sure that javadoc is still deployed to https://docs.langchain4j.dev/apidocs/index.html

**Review by ghost — DISMISSED** [review]
@kpavlov thank you!

**Comment by langchain4j** [comment]
@kpavlov would you like to add a documentation page for Kotlin?

**Comment by kpavlov** [comment]
> @kpavlov would you like to add a documentation page for Kotlin?

Sure, I will add it

**Comment by langchain4j** [comment]
> Sure, I will add it

In scope of this PR or should I merge and you will add it later?

File: langchain4j-core/pom.xml (hunk 1 above)
Comment (kpavlov) [reply langchain4j-core/pom.xml]: I see it is still generated under `target/site/apidocs` with `mvn -T4C clean compile javadoc:aggregate`

**Comment by kpavlov** [comment]
> In scope of this PR or should I merge and you will add it later?

I prefer separate PR

**Comment by kpavlov** [comment]
Added documentation here

**Review by ghost — APPROVED** [review]
@kpavlov thanks a lot!

**Pull request merged**